### PR TITLE
Create factory contract for phase0

### PR DESCRIPTION
**Pull request by CodeSandwich**
Due to what I believe is a hardhat bug (https://github.com/nomiclabs/hardhat/issues/1227) a few tests aren't verifying transaction rejection messages anymore.

### CODE REVIEW

File: contracts/deploy/phase0.sol [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+        );
+        uint8 governorNonce = 3;
+        bytes memory govAddrPayload = abi.encodePacked(hex"d694", address(this), governorNonce);
+        address govAddr = address(uint256(keccak256(govAddrPayload)));
+
+        RadicleToken _token = new RadicleToken(_tokensHolder);
+        Timelock _timelock = new Timelock(govAddr, _timelockDelay);
+        Governor _governor = new Governor(address(_timelock), address(_token), _governorGuardian);
+        require(address(_governor) == govAddr, "Governor deployed under an unexpected address");
+        Treasury _treasury = new Treasury(address(_timelock));
```
Comment (cloudhead) [inline contracts/deploy/phase0.sol:47]: Can we actually use the Timelock already here?

Comment (CodeSandwich) [reply contracts/deploy/phase0.sol]: Fixed, just removed treasury deployment, we don't need anything else done

File: contracts/deploy/phase0.sol [hunk 2]
```diff
@@ -13,7 +13,6 @@ contract Phase0 {
     RadicleToken public immutable token;
```
Comment (cloudhead) [inline contracts/deploy/phase0.sol:13]: Code is correct, but commit message isn't ;)

Comment (CodeSandwich) [reply contracts/deploy/phase0.sol:13]: What do you mean?

Comment (cloudhead) [reply contracts/deploy/phase0.sol:13]: It says "Remove timelock deployment"

Comment (CodeSandwich) [reply contracts/deploy/phase0.sol:13]: Yes and the commit with that message does exactly that :thinking: Anyway we shouldn't sweat too much over any commit messages except the first one, they will be squashed before merging.

Comment (cloudhead) [reply contracts/deploy/phase0.sol:13]: The commit removes the treasury, not the timelock :grimacing:

**Review by cloudhead — APPROVED** [review]

File: contracts/deploy/phase0.sol (hunk 2 above)
Comment (CodeSandwich) [reply contracts/deploy/phase0.sol:13]: Right :facepalm:

**Pull request merged**